internal/controller/eventing/unit_test.go

```diff
@@ -61,11 +63,14 @@ func NewMockedUnitTestEnvironment(t *testing.T, objs ...client.Object) *MockedUn
 	err = admissionv1.AddToScheme(newScheme)
 	require.NoError(t, err)
 
+	// Create a fake dynamic client
+	dynamicClient := fakedynamic.NewSimpleDynamicClient(newScheme)
```

Comment:
nit: to be consistent with the rest of the code, maybe rename to `fakeDynamicClient`